### PR TITLE
Remove unused variables from Terraform Scripts which are shown as Warnings in output

### PR DESCRIPTION
**Pull request by bvtejaswi**
Signed-off-by: Tejaswi Bondila <bondila.venkatatejaswi@progress.com>

### :nut_and_bolt: Description: What code changed, and why?
Removed unused variables from Terraform scripts and declared already using variables to avoid Warning or Error related to Unused or Unset Terraform Variables.

### :chains: Related Resources
https://chefio.atlassian.net/browse/MADROX-437

### :+1: Definition of Done

- Tested Onprem deployment with chef managed, AWS managed and self managed services and also  AWS deployment with chef managed and AWS managed services

In all of the above scenarios the testing flow is as following:

1. Deployed with 4.2.59 version( for onprem aws and self managed the deployment is done with 4.4.10)
2. Upgraded to 4.5.34 version.
3. Added and removed  a2 and cs nodes.
4. Then upgraded to latest dev.
5. Added and Removed a node again.

Onprem Deployment output without changes
https://progresssoftware-my.sharepoint.com/:i:/g/personal/btejaswi_progress_com/EWXLSdb740NOgXPdXqVDmqAB6BxKniiP0Hq8C6TEdWc3YQ?e=GF45IR
Onprem Deployment output with changes
https://progresssoftware-my.sharepoint.com/:i:/g/personal/btejaswi_progress_com/EfDhcyn4LZxCukyv1KUoJTMBf4_XtxkLXlqa5quuWbsj-g?e=kFRBxG
AWS Deployment output without changes
https://progresssoftware-my.sharepoint.com/:i:/g/personal/btejaswi_progress_com/EWlJf5BQIwFLl-KGLP6JdDEBNa4wqUtlpE6p1uQ83PZRyw?e=mHM5eB
AWS Deployment output with changes
https://progresssoftware-my.sharepoint.com/:i:/g/personal/btejaswi_progress_com/EQKqFVuV3eNHo8KZaKXL2vsBMg5b2XxZGad1oB3BP-Vz6Q?e=NxXiue

### :athletic_shoe: How to Build and Test the Change
To test this change you should build `automate-cli`,`automate-cluster-ctl` and`automate-ha-deployment`

### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

**Review by punitmundra — APPROVED** [review]

**Review by jayvikramsharma1 — APPROVED** [review]

File: components/automate-cli/cmd/chef-automate/ha_node_operations_utils.go [hunk 1]
```diff
@@ -217,9 +217,11 @@ func (nu *NodeUtilsImpl) isA2HARBFileExist() bool {
 }
 
 func (nu *NodeUtilsImpl) taintTerraform(path string) error {
+	modeOfDeployment := getModeOfDeployment()
+	if modeOfDeployment == AWS_MODE {
 	if err := executeShellCommand("/bin/bash", []string{"-c", REMOVE_TERRAFORM_STATE}, filepath.Join(path, "destroy", "aws")); err != nil {
 		return err
-	}
+	}}
```
Comment (vivek-yadav) [inline components/automate-cli/cmd/chef-automate/ha_node_operations_utils.go:224]: indentation seems not correct, please use gofmt in your vscode plugin to run every time you save go file

File: terraform/a2ha-terraform/variables_common.tf [hunk 2]
```diff
@@ -38,6 +38,11 @@ variable "automate_fqdn" {
   description = "Automate FQDN variable. Leave as an empty string for AWS"
 }
 
+variable "automate_frontend_urls" {
```
Comment (vivek-yadav) [inline terraform/a2ha-terraform/variables_common.tf:41]: why is this value plural ? 
```suggestion
variable "automate_frontend_url" {
```

Comment (bvtejaswi) [reply terraform/a2ha-terraform/variables_common.tf:41]: In terraform where ever we are using this variable it was already used as "automate_frontend_urls",if you want me to change as "automate_frontend_url" then i need to change in the existing files as well if thats ok I can change it.

File: components/automate-cli/cmd/chef-automate/ha_node_operations_utils.go (hunk 1 above)
Comment (bvtejaswi) [reply components/automate-cli/cmd/chef-automate/ha_node_operations_utils.go:224]: sure

File: terraform/a2ha-terraform/variables_common.tf (hunk 2 above)
Comment (vivek-yadav) [reply terraform/a2ha-terraform/variables_common.tf:41]: Yes please replace it **carefully**.

Comment (vivek-yadav) [reply terraform/a2ha-terraform/variables_common.tf:41]: We are sure that it only stores one value and not multiple values?

**Review by jayvikramsharma1 — APPROVED** [review]

**Review by vivek-yadav — APPROVED** [review]

File: terraform/a2ha-terraform/modules/aws/main.tf [hunk 3]
```diff
@@ -207,9 +207,9 @@ resource "aws_instance" "chef_automate_postgresql" {
   )
   lifecycle {
     ignore_changes = [
-      "tags",
-      "tags_all",
-      "root_block_device"
+      tags,
```
Comment (ArvinthC3000) [inline terraform/a2ha-terraform/modules/aws/main.tf:210]: Why the string is replaced?

**Review by ArvinthC3000 — APPROVED** [review]

**Pull request merged**